### PR TITLE
Create Plugin Modal

### PR DESCRIPTION
**Pull request by KTS915**
This PR creates a modal dialog for when the "More Details" link is clicked on the CP Plugins page.

In order to make this easier to add to core in future, I have tried, as much as possible, to use CSS class names and styles that CP already uses in the back-end. I have also followed the rules we use for core JavaScript, which is why I have used `var` instead of `const` and `let`.

One thing I am not sure about is whether I have used `wp_set_script_translations` correctly on line 37 of `PluginInstall.class.php`.

### CODE REVIEW

File: classes/PluginInstall.class.php [hunk 1]
```diff
@@ -25,15 +25,16 @@ public function styles($hook)
 		if ($hook !== $this->page) {
 			return;
 		}
-		wp_enqueue_style('classicpress-directory-integration-css-plugin', plugins_url('../styles/plugin-page.css', __FILE__), []);
+		wp_enqueue_style('classicpress-directory-integration-css', plugins_url('../styles/plugin-page.css', __FILE__), []);
```
Comment (xxsimoxx) [inline classes/PluginInstall.class.php:28]: In the last PR `-plugin` was added to have an handle that differs from the `-theme` one.

File: classes/PluginInstall.class.php [hunk 2]
```diff
@@ -25,15 +25,16 @@ public function styles($hook)
...
 	}
 
 	public function scripts($hook)
 	{
 		if ($hook !== $this->page) {
 			return;
 		}
-		wp_enqueue_script('classicpress-directory-integration-js-plugin', plugins_url('../scripts/plugin-page.js', __FILE__), ['jquery'], false, true);
+		wp_enqueue_script( 'classicpress-directory-integration-js', plugins_url( '../scripts/plugin-page.js', __FILE__ ), array( 'wp-i18n' ), false, true );
```
Comment (xxsimoxx) [inline classes/PluginInstall.class.php:36]: In the last PR `-plugin` was added to have an handle that differs from the `-theme` one.

**Comment by xxsimoxx** [comment]
Apart from the two annotations, it looks good!

The same thing should be done for the new theme install page, along with the same changes made in #15.
Do you think is better doing in another PR?

File: classes/PluginInstall.class.php (hunk 1 above)
Comment (KTS915) [reply classes/PluginInstall.class.php]: Now done.

**Comment by KTS915** [comment]
Yes, I'll do another PR for themes. Probably tomorrow.

File: classes/PluginInstall.class.php [hunk 3]
```diff
@@ -33,7 +33,8 @@ public function scripts($hook)
 		if ($hook !== $this->page) {
 			return;
 		}
-		wp_enqueue_script('classicpress-directory-integration-js-plugin', plugins_url('../scripts/plugin-page.js', __FILE__), ['jquery'], false, true);
+		wp_enqueue_script( 'classicpress-directory-integration-js-plugin', plugins_url( '../scripts/plugin-page.js', __FILE__ ), array( 'wp-i18n' ), false, true );
+		wp_set_script_translations( 'classicpress-directory-integration-js', 'classicpress-directory-integration', plugin_dir_path( 'classicpress-directory-integration' ) . 'languages' );
```
Comment (xxsimoxx) [inline classes/PluginInstall.class.php:37]: I think you have to change the handle also here:
`wp_set_script_translations( 'classicpress-directory-integration-js-plugin', ....`

Comment (KTS915) [reply classes/PluginInstall.class.php]: Good spot! Now done.

**Review by xxsimoxx — APPROVED** [review]
Looks good!

**Pull request merged**